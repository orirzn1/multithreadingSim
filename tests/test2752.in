L12
S8
O6
N14

T0
I@0x00000000
SUB $0, $3, $4
LOAD $6, $1, $7
LOAD $3, $1, $5
ADDI $1, $6, 0x3
ADD $0, $6, $5
LOAD $4, $7, $0
HALT $0

T1
I@0x00000000
SUB $6, $2, $1
SUB $2, $5, $3
LOAD $4, $2, $3
SUB $6, $4, $5
STORE $5, $7, $2
LOAD $0, $6, 0x4
HALT $0

T2
I@0x00000000
ADD $5, $7, $1
SUB $7, $3, $7
LOAD $2, $5, 0x2
SUBI $0, $2, 0x2
ADD $2, $3, $6
SUB $7, $3, $4
ADDI $7, $3, 0x4
SUBI $6, $5, 0x4
HALT $0

T3
I@0x00000000
ADDI $6, $3, 0x1
LOAD $5, $1, $7
STORE $4, $7, $2
SUB $6, $3, $4
ADD $3, $5, $3
STORE $7, $4, 0x4
HALT $0

T4
I@0x00000000
ADDI $2, $0, 0x2
LOAD $5, $0, $1
SUB $0, $4, $5
ADDI $2, $5, 0x1
ADD $3, $4, $3
SUBI $6, $5, 0x4
ADDI $3, $5, 0x4
LOAD $3, $6, 0x3
ADDI $5, $4, 0x3
HALT $0

T5
I@0x00000000
SUB $5, $3, $7
STORE $0, $5, 0x0
ADDI $2, $4, 0x1
ADDI $4, $5, 0x4
SUBI $7, $7, 0x1
LOAD $4, $5, 0x4
LOAD $5, $2, $4
STORE $5, $3, $4
SUB $1, $3, $3
SUB $2, $3, $7
HALT $0

T6
I@0x00000000
SUB $5, $1, $5
ADD $5, $0, $5
SUBI $7, $2, 0x2
SUBI $5, $0, 0x4
LOAD $1, $0, $2
SUB $7, $0, $4
HALT $0

T7
I@0x00000000
STORE $4, $3, $7
STORE $7, $2, $5
LOAD $0, $1, 0x2
ADDI $0, $6, 0x4
HALT $0

T8
I@0x00000000
ADDI $1, $2, 0x4
STORE $7, $2, 0x1
LOAD $0, $3, $3
LOAD $1, $0, 0x1
SUBI $0, $2, 0x3
ADD $0, $6, $5
HALT $0

T9
I@0x00000000
ADDI $6, $4, 0x3
SUBI $7, $2, 0x2
ADD $1, $5, $1
ADDI $2, $6, 0x1
ADDI $7, $7, 0x3
SUBI $6, $2, 0x4
LOAD $6, $6, 0x3
HALT $0

T10
I@0x00000000
ADD $5, $5, $6
SUB $3, $7, $0
ADDI $1, $6, 0x2
ADDI $2, $6, 0x3
ADDI $0, $3, 0x1
ADDI $6, $7, 0x1
HALT $0

T11
I@0x00000000
ADDI $7, $4, 0x3
ADDI $1, $2, 0x1
ADD $6, $4, $4
STORE $1, $0, 0x0
SUB $1, $5, $0
ADDI $2, $3, 0x3
SUBI $7, $2, 0x2
SUBI $4, $5, 0x1
STORE $4, $5, $2
LOAD $5, $7, $2
HALT $0

T12
I@0x00000000
LOAD $6, $0, $4
LOAD $6, $3, $5
ADDI $3, $4, 0x4
ADDI $4, $7, 0x1
SUBI $4, $5, 0x3
SUBI $2, $4, 0x1
SUBI $0, $3, 0x4
ADD $3, $5, $1
SUBI $0, $3, 0x3
HALT $0

T13
I@0x00000000
LOAD $5, $3, 0x0
SUB $1, $5, $6
STORE $2, $0, 0x4
LOAD $2, $6, 0x4
SUB $5, $2, $6
LOAD $2, $4, 0x3
ADDI $5, $3, 0x3
HALT $0

D@0x0
0x3
0x2
0x3
0x3
0x4
0x8
