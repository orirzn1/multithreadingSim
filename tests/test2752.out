
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x3	R1 = 0x6	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x3	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 2:
	R0 = 0x1	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0x4
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x3	R4 = 0x0	R5 = 0x3	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0xFFFFFFFF	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x2	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x1
Register file thread id 10:
	R0 = 0x1	R1 = 0x2	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x1	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0xFFFFFFFF	R5 = 0x0	R6 = 0x0	R7 = 0x1
Register file thread id 12:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 3.447368

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x3	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0xFFFFFFFE	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0x4
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0xFFFFFFFC	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFF
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFC	R6 = 0x0	R7 = 0x0
Register file thread id 7:
	R0 = 0x4	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 8:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x1
Register file thread id 10:
	R0 = 0x1	R1 = 0x2	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x1	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0xFFFFFFFF	R5 = 0x0	R6 = 0x0	R7 = 0x1
Register file thread id 12:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 13:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x3	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.105263

